File: 713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        s=0
        e=0
        p=1
        cnt=0
        while e<len(nums):
            p=p*nums[e]
            if p<k:
                cnt=cnt+(e-s+1)
                e=e+1
            else:
                while p>=k and s<=e:
                    p=p/nums[s]
                    s=s+1
                if p<k:
                    cnt=cnt+(e-s+1)
                e=e+1
        return cnt

        

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int left=0;
        int right=0;
        int length=nums.length;
        int count=0;
        int product=1;
        while(right<length){
            product=product*nums[right];
            while(product>=k && left<right){
                product=(product/nums[left]);
                left=left+1;
            }
            if(product<k){
                count=count+(right-left+1);
            }
            right++;
        } 
        return count;
    }
}